### DISCUSSION TITLE
Full name of some variable and function names

### DISCUSSION THREAD

**Discussion by JavanZhu** [bytecodealliance/wasm-micro-runtime#578] [Q&A] [answered]
1. In file ems_kfc.c, what is the full name of "ems" and "kfc"?
2. And in "read_leb", what is the full name of "leb"?
https://github.com/bytecodealliance/wasm-micro-runtime/blob/fda3a26903512c13078f47e7c9e082f1b2c7c794/core/iwasm/interpreter/wasm_loader.c#L87-L96
3. In "frame_ip", not sure the full name of "ip" whether is the "instruction pointer", and some other similar variable names like "frame_lp", "frame_sp", "frame_csp" in the same function, also curious about the full name of "lp" and "sp" and "csp".
https://github.com/bytecodealliance/wasm-micro-runtime/blob/fda3a26903512c13078f47e7c9e082f1b2c7c794/core/iwasm/interpreter/wasm_interp_classic.c#L1086-L1089

please tell me if anyone knows, thank you very much.

**Comment by kfkfkfkfkfkf** [comment, marked as answer]
1. Sorry, I don't know too.
2. "leb" is "leb128", "Little Endian Base 128".
3. frame_ip is "instruction pointer".frame_sp is "stack pointer".frame_lp is probably "local pointer".I don't know csp,sorry.

I wanna know what is the full name of "hmu" and "gc".

**Reply by JavanZhu** [reply]
Hi @kan-fumihito, Thanks for your reply. The full name of "hmu" is "heap memory unit" which could be referenced at 
https://github.com/bytecodealliance/wasm-micro-runtime/blob/6b3a93937083d4c2bf3871d469b1df251248efa3/core/shared/mem-alloc/ems/ems_gc_internal.h?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L14-L17
and maybe the "gc" is abbreviation of "garbage collection".
